src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -273,16 +277,36 @@ def csv(self, path: str) -> DataFrame:
 
         self._file_path = path
         self._file_type = "csv"
-        df = DataFrame(
-            self._session,
-            self._session._plan_builder.read_file(
-                path,
-                self._file_type,
-                self._cur_options,
-                self._session.get_fully_qualified_current_schema(),
-                self._user_schema._to_attributes(),
-            ),
-        )
+        from snowflake.snowpark import context
+
+        if context._use_sql_simplifier:
```

Comment:
Again, may be out of scope from this PR but why import `snowflake.snowpark.context` locally near the simplifier change instead of at the top of file?

Reply:
It would have circular import error. I'm thinking of a way to improve it. Will discuss with the team.

Reply:
We should probably prioritize this. The later we fix this the harder I think.